### DISCUSSION TITLE
PeriodTypeSelector, how to change value programmatically

### DISCUSSION THREAD

**Discussion by PavloParafiloCricut** [Megabit/Blazorise#4355] [Q&A] [answered]
If try to change it via component reference Dates, it is some warning. 
So how to change/set values from codebehind?
![image](https://user-images.githubusercontent.com/96420517/205284439-6fac2a88-9e3d-477b-8121-df8cf60fcf80.png)

**Comment by stsrki** [comment, marked as answer]
You're not supposed to change parameters directly. That warning comes from Blazor and warns you not to do it. Remember, Blazor, like most other SPA frameworks, prefer a declarative approach. That means you need to set parameters through the razor syntax.

Pseudo sample code:

```razor
<DatePicker Dates="@someDates" ... />

Task OnSomeAction()
{
  someDates = newDates;

  return Task.CompletedTask;
}
```

**Reply by PavloParafiloCricut** [reply]
Thanks for the answer. But Dates is IReadonly list. I am trying to set like below and getting some strange behaviour: it set's dates perfectly when second block of code is called, but when I click on any date in calendar - it trigers change event but with old values (not what I selected on UI). I will play with it more. But probably any advise in advance?

```
        <DatePicker TValue="DateOnly?"
                    Width="Width.Is100"
                    DatesChanged="OnDateRangeFilterChanged"
                    Dates="_dateRangeFilterValue"
                    Placeholder="Select range..."
                    @ref="_dateRangeFilterRef"
                    SelectionMode="DateInputSelectionMode.Range" />
```


```
_dateRangeFilterValue.Clear();

            switch (periodType)
            {
                case PeriodType.Day:
                    _dateRangeFilterValue.AddRange(new List<DateOnly?>
                    {
                        DateOnly.FromDateTime(DateTime.Now.GetPreviousWeekDay(DayOfWeek.Monday)),
                        DateOnly.FromDateTime(DateTime.Now.GetPreviousWeekDay(DayOfWeek.Friday))
                    });
                    break;
                case PeriodType.Week:
                case PeriodType.Month:
                    _dateRangeFilterValue.AddRange(new List<DateOnly?>
                    {
                        DateOnly.FromDateTime(DateTime.Now.AddYears(-1)),
                        DateOnly.FromDateTime(DateTime.Now)
                    });
                    break;
                default:
                    throw new ArgumentException(nameof(periodType));
            }

            return Task.CompletedTask;
```

**Reply by PavloParafiloCricut** [reply]
It doesn't work even for simple example. Created a ticked,  please take a look https://github.com/Megabit/Blazorise/issues/4356
